graphite/graphite.go

```diff
@@ -0,0 +1,188 @@
+package graphite
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdb/influxdb"
+)
+
+var (
+	// ErrBindAddressRequired is returned when starting the Server
+	// without a TCP or UDP listening address.
+	ErrBindAddressRequired = errors.New("bind address required")
+
+	// ErrServerClosed return when closing an already closed graphite server.
+	ErrServerClosed = errors.New("server already closed")
+
+	// ErrDatabaseNotSpecified retuned when no database was specified in the config file
+	ErrDatabaseNotSpecified = errors.New("database was not specified in config")
+
+	// ErrServerNotSpecified returned when Server is not specified.
+	ErrServerNotSpecified = errors.New("server not present")
+)
+
+type ProcessorSink interface {
+	WriteSeries(database, retentionPolicy, name string, tags map[string]string, timestamp time.Time, values map[string]interface{}) error
+	DefaultRetentionPolicy(database string) (*influxdb.RetentionPolicy, error)
+}
+
+// Processor performs processing of Graphite data, and writes it to a sink.
+type Processor struct {
+	// Sink is the destination for processed Graphite data.
+	sink dataSink
+
+	// Database is the name of the database to insert data into.
+	Database string
+	// NamePosition is the position of name to be parsed from metric_path
+	NamePosition string
+	// NameSeparator is separator to parse metric_path with to get name and series
+	NameSeparator string
+}
```

Comment:
I think adding a whole type to handle messages is overkill. The fields are simple enough that you could just copy them to both the `TCPServer` and `UDPServer`. Having sinks & processors and servers seems like a lot of complexity for this code.